py/server/deephaven/experimental/iceberg.py

```diff
@@ -264,58 +265,96 @@ def adapter(
             `org.apache.iceberg.aws.glue.GlueCatalog`). Choices for `type` include `hive`, `hadoop`, `rest`, `glue`,
             `nessie`, `jdbc`.
 
+    If the user is responsible for providing AWS / S3 connectivity details, setting s3_instructions is the recommended
+    way of providing these details. This allows for the parity of construction logic between Iceberg-managed and
+    Deephaven-managed AWS clients. For advanced use-cases, users are encouraged to use S3Instruction profiles which
+    allows a rich degree of configurability. When set, these instructions will also be automatically included as part of
+    IcebergInstructions data_instructions.
+
+    In cases where the caller prefers to use Iceberg's AWS properties, the parity of construction logic will be limited
+    to what Deephaven is able to infer; in advanced cases, it's possible that there will be a difference in construction
+    logic between the Iceberg-managed and Deephaven-managed AWS clients which manifests itself as being able to browse
+    Catalog metadata, but not retrieve Table data.
+
```

Comment:
A little unclear without an example. Also it might be nice to point out a possible solution for the problem of not being able to read table data.

Reply:
The problem is that it's a very opened thing, and it's tough to provide generalizabled example.